lib/sdk.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2016 Kinvey Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+'use strict';
+
+const dataLink = require('./service/dataLink');
+const businessLogic = require('./service/businessLogic');
+const moduleGenerator = require('./service/moduleGenerator');
+
+const receiver = require('kinvey-code-task-runner');
+
+const generateService = (callback) => {
+  const taskReceivedCallback = (task, completionCallback) => {
+    if (task.taskType === 'dataLink') {
+      return dataLink.process(task, moduleGenerator.generate(task), completionCallback);
+    } else if (task.taskType === 'businessLogic') {
+      return businessLogic.process(task, moduleGenerator.generate(task), completionCallback);
+    }
+  };
+
+  receiver.start(taskReceivedCallback, (err) => {
+    if (typeof err !== 'undefined' && err !== null) {
+      return callback(new Error(`Could not start task receiver: ${err}`));
+    }
+    return callback(null, this);
+  });
+};
+
+exports.service = (callback) => { generateService((err, service) => { callback(err, service); }); };
```

Comment:
Does this work?  In the CoffeeScript it was an es6 class which had reference to `dataLink`, `businessLogic`, and `moduleGenerator` methods.  That's how a user defined their datalink.  

Why not keep this an es6 class in a similar manner?